repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,274 @@
+    help="An artifacts base URL to fetch from.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.option(
+    "-p",
+    "--hash-prefix",
+    help="A flag to prefix an artifact with a hash.",
+    is_flag=True,
+    required=None,
+)
+@click.option(
+    "-P",
+    "--directory-prefix",
+    help="A prefix for the download dir.",
+    type=str,
+    required=None,
+    default=None,
+)
+@click.argument("artifact_name")
+@click.pass_context
+def download(
+    context: Context,
+    root: Optional[str],
+    metadata_url: Optional[str],
+    artifacts_url: Optional[str],
+    hash_prefix: Optional[bool],
+    directory_prefix: Optional[str],
+    artifact_name: str,
+) -> None:
+    """
+    Downloads artifacts to the TUF metadata.
+    """
+
+    rstuf_config = context.obj.get("settings")
+
+    if context.obj.get("config") and os.path.isfile(context.obj.get("config")):
+        repository = rstuf_config.get("CURRENT_REPOSITORY")
+        # Overwriting the config if a flag is passed
+        # or using it as default instead
+        if not repository:
+            raise click.ClickException("Please specify current repository")
+        if not rstuf_config.get("REPOSITORIES"):
+            raise click.ClickException(
+                "No reposotiroes listed in the config file"
+            )
+        if not rstuf_config["REPOSITORIES"].get(repository):
+            raise click.ClickException(
+                f"Repository {repository} is missing in the "
+                "configuration file"
+            )
+        if not artifacts_url:
+            artifacts_url = rstuf_config["REPOSITORIES"][repository].get(
+                "artifact_base_url"
+            )
+        if not metadata_url:
+            metadata_url = rstuf_config["REPOSITORIES"][repository].get(
+                "metadata_url"
+            )
+        if not hash_prefix:
+            hash_prefix = rstuf_config["REPOSITORIES"][repository].get(
+                "hash_prefix", False
+            )
+        if not directory_prefix:
+            directory_prefix = rstuf_config["REPOSITORIES"][repository].get(
+                "download_dir"
+            )
+        if not root:
+            root = rstuf_config["REPOSITORIES"][repository].get("trusted_root")
+            if not root:
+                console.print(
+                    "Trusted root is not cofigured. "
+                    "You should either add it to your config file, "
+                    "or use the download commang without a config file"
+                )
+                return
+            root = decode_trusted_root(root)  # pragma: no cover
+            console.print(f"Decoded trusted root {root}")
+
+    ok = _download_artifact(
+        metadata_url,
+        artifacts_url,
+        hash_prefix,
+        directory_prefix,
+        artifact_name,
+        root,
+    )
+
+    if ok:
+        console.print(
+            f"Successfully completed artifact download: {artifact_name}"
+        )
```

Comment:
No need of parsing any boolean result as I mentioned above.
If it reached the next line after `_download_artifact` its clear it didn't raise.